docs/proposals/13-automated-seed-management.md

```diff
@@ -0,0 +1,99 @@
+
+The next step in migrating shoots away from an unhealthy seed is rescheduling all orphaned shoots to healthy seeds by `gardener-scheduler`. The scheduler should ensure, among other things, that:
+
+* A shoot with the `orphan` annotation or with an empty `seedName` field is rescheduled.
+* A shoot is never scheduled on a seed tainted with a taint with a `NoSchedule` effect (unless the shoot tolerates the taint).
+* The new seed capacity for shoots is not exceeded by adding a new shoot, see [Ensuring Seeds Capacity for Shoots Is Not Exceeded](#ensuring-seeds-capacity-for-shoots-is-not-exceeded). This could perhaps also be achieved by automatically tainting the seed with a `NoSchedule` taint once its capacity is exceeded by the `gardenlet`.
+
+To reschedule a shoot, the scheduler updates the `seedName` field in the shoot specification with the name of the new seed. 
+
+### Migrating the Shoots to the New Seeds
+
+The final step in migrating shoots away from an unhealthy seed is actually migrating the shoots control planes from the old to the new seeds. The migration is triggered by the updates to the `seedName` field in the shoot specification by the scheduler described above.
+
+The migration process is described in more detail in [GEP 7: Shoot Control Plane Migration](07-shoot-control-plane-migration.md). 
+
+## Ensuring Seeds Capacity for Shoots Is Not Exceeded
+
+Another important automated seed management scenario is ensuring that a seed's capacity for shoots is not exceeded by introducing a maximum number of shoots that can be scheduled on a seed and making sure that it is taken into account by the scheduler. 
+
+An initial discussion of this topic is available in [Max cap for shoots per seed in scheduler](https://github.com/gardener/gardener/issues/2938). The proposed solution is based on the following flow:
+
+* The `gardenlet` is configured with certain *resources* and their *capacity*.
+* The `gardenlet` seed controller updates the seed status with the capacity of each resource and how much of it is currently available, using `capacity` and `allocatable` fields that are very similar to the corresponding fields in [the Node status](https://github.com/kubernetes/api/blob/2c3c141c931c0ab1ce1396c3152c72852b3d37ee/core/v1/types.go#L4582-L4593).
+* When scheduling shoots, `gardener-scheduler` is influenced by the allocatable vs. capacity score of the seed. In the simplest possible implementation, it never schedules shoots to a seed that has already reached its maximum capacity for a resource requested by the shoot.
+
+Initially, the only resource considered would be the maximum number of shoots that can be scheduled on a seed. Later, more resources could be added, as well as requests for such resources by shoots. For example, if a shoot requests (via a `resources` field) a certain number of persistent volumes, but the seed's `allocatable` value for persistent volumes is lower, the shoot can't be scheduled on that seed.
```

Comment:
In the future, seed cluster _health_ could also be a criteria during scheduling (perhaps mediated by taints as you have proposed).

Reply:
Perhaps, but following the team discussion I removed the mentioning of `taints` from this GEP. I will file another PR with this to keep the topic alive, but I don't plan to actively push it through for the time being.